### PR TITLE
fix raising error during cli lazy loading

### LINKED ISSUE #4096: Exceptions are sometimes replaced with "TypeError: exceptions must derive from BaseException"
```python
# a.py
def create_app(): raise RuntimeError()
```
```
$ FLASK_APP=a.py flask run --lazy-loading
$ curl http://127.0.0.1:5000
[...]
Traceback (most recent call last):
  File "[...]/lib/python3.9/site-packages/flask/cli.py", line 356, in __call__
    self._flush_bg_loading_exception()
  File "[...]/lib/python3.9/site-packages/flask/cli.py", line 344, in _flush_bg_loading_exception
    raise exc_info
TypeError: exceptions must derive from BaseException
```

I expected something about a RuntimeError.  `raise exc_info[1]` here worked for me https://github.com/pallets/flask/blob/7161776824734fc2797fe2b4fc974d183487ebf8/src/flask/cli.py#L342

Environment:

- Python version: 3.9.5
- Flask version: 2.0.1

### PR DESCRIPTION
**Pull request by Rohan-Salwan**
- fixes #4096

Checklist:

- [x] I did tests that demonstrate the correct behavior of the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [ ] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

**Comment by davidism** [comment]
I don't think this is correct. `raise sys.exc_info()` should be valid, so discarding everything but the exception instance shouldn't be required, and may be discarding rewritten tracebacks in some cases.

You need to identify what causes `exc_info` to be invalid, add a test case for that, and address that specific issue.

**Comment by davidism** [comment]
After investigating a bit more, this appears to be a leftover from Python 2, which did support `raise sys.exc_info()`. That's no longer necessary, and doesn't appear to have ever been necessary, since we weren't replacing the traceback.

Given that, I think we should stop storing `exc_info` and only store the exception instead. Rename `self._bg_loading_exc_info` to `_bg_loading_exc` to reflect that as well.

**Comment by Rohan-Salwan** [comment]
Hi David,
I did some code changes please check that and if I missed some thing then please reply. Thanks for your time

**Comment by davidism** [comment]
Code looks good. Needs a changelog, and if possible a test.

**Comment by Rohan-Salwan** [comment]
hey @davidism, 
its done please review it.

File: CHANGES.rst [hunk 1]
```diff
@@ -20,6 +20,9 @@ Version 2.0.1
 
 Released 2021-05-21
 
+-   Renamed a variable self._bg_loading_exc_info to self._bg_loading_exc
+    and used that variable to store Exception instead of storing
+    Exception info and also wrote Exception testcase for it. :pr:`4169`
```
Comment (ThiefMaster) [inline CHANGES.rst:25]: Don't add changelog entries for an already-released version... add them for 2.0.2 (at the end of that list)

Comment (Rohan-Salwan) [reply CHANGES.rst]: Done

**Pull request merged**